ptrail/regularization/goemetric_chull_related.py

```diff
@@ -0,0 +1,279 @@
+
+    Returns:
+        np.ndarray: Array of major axes distances for each point in the convex hull.
+    """
+    try:
+        return np.apply_along_axis(lambda vertex: distance.euclidean(vertex, chull_centroid), 1, chull)
+    except Exception as error:
+        # Handle the exception (e.g., print an error message, return a default value, etc.)
+        print("An error occurred:", str(error))
+        return np.array([])  # Return an empty array or any other default value
+
+
+def major_axis_wrapper(chull: np.ndarray, chull_centroid: np.ndarray) -> Union[np.ndarray, float]:
+    """
+    Wraps the calculation of major axis distances for a convex hull and handles exceptions.
+
+    Args:
+        chull (np.ndarray): Array of points representing the convex hull.
+        chull_centroid (np.ndarray): Array of coordinates representing the centroid of the convex hull.
+
+    Returns:
+        Union[np.ndarray, float]: Array of major axis distances for each point in the convex hull if successful,
+        otherwise returns NaN if an exception occurs.
+    """
+    try:
+        return np.apply_along_axis(lambda vertex: np.linalg.norm(vertex - chull_centroid), 1, chull)
+    except Exception as error:
+        return np.nan
+
+
+# Calculate major axis distances for each convex hull
+chulls_major_axes = [major_axis_wrapper(
+    chull, centroid) for chull, centroid in zip(chulls, chulls_centroids)]
+
+# Calculate the maximum major axis distance for each convex hull
+chulls_major_axis = [np.max(distances) for distances in chulls_major_axes]
+
+# Shift convex hulls by removing the first point and appending the first point at the end
+chulls_shifted = [chull[1:] for chull in chulls]
+chulls_shifted = [np.vstack((shifted, chull[0]))
+                  for shifted, chull in zip(chulls_shifted, chulls)]
+
+# Convert convex hulls and shifted hulls to data frames
+chulls_df = [pd.DataFrame(chull, columns=['x', 'y']) for chull in chulls]
+chulls_shifted_df = [pd.DataFrame(shifted, columns=['x', 'y'])
+                     for shifted in chulls_shifted]
+
+# Rename columns in chulls_shifted_df
+chulls_shifted_df = [
+    df.rename(columns={'x': 'V3', 'y': 'V4'}) for df in chulls_shifted_df]
+
+# Combine convex hull data frames and shifted data frames
+chulls_traj_bounding_points = [pd.concat(
+    [df1, df2], axis=1) for df1, df2 in zip(chulls_df, chulls_shifted_df)]
+
+# Convert centroids to data frames
+chulls_centroids_df = [pd.DataFrame(np.transpose(centroid), columns=[
+                                    'V5', 'V6']) for centroid in chulls_centroids]
+
+# Rename columns in chulls_centroids_df
+chulls_centroids_df = [
+    df.rename(columns={'V1': 'V5', 'V2': 'V6'}) for df in chulls_centroids_df]
+
+# Combine chulls_traj_bounding_points and chulls_centroids_df
+chulls_traj_bounding_points_and_centroid = [pd.concat(
+    [df1, df2], axis=1) for df1, df2 in zip(chulls_traj_bounding_points, chulls_centroids_df)]
+
```

Comment:
Do not do testing like this inline in the source code file, this is not what is done in the library. Please see how other modules are coded and tested. Make a new jupyter notebook show all your testing there.